mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/SocketPolicy.java

```diff
@@ -50,6 +50,9 @@
    */
   DISCONNECT_AFTER_REQUEST,
 
+  /** Close connection after reading half of the request body (if present). */
```

Comment:
Should I add a warning about the asynchronous notification of close which might not result in failure on small bodies?